.travis.yml

```diff
@@ -2,30 +2,41 @@ branches:
   only:
   - master
 language: groovy
-sudo: false
-install:
-- pip install --user mkdocs==0.17.5 mkdocs-material==2.9.4
+sudo: required
+services:
+ - docker
+before_install:
+- docker pull squidfunk/mkdocs-material
 script:
 - mvn clean test --batch-mode
 - |
   if [[ "${TRAVIS_PULL_REQUEST}" != "false" ]]
   then
-    cd documentation && mkdocs build --clean --verbose --strict && cd ..
+    cd documentation && docker run --rm -it -v ${PWD}:/docs squidfunk/mkdocs-material build --clean --verbose --strict && cd ..
   else
     # Only in case we are in master branch of the leading SAP repo we would like to deploy,
     # not from the forks.
     if [[ "${TRAVIS_BRANCH}" == "master" && "${TRAVIS_REPO_SLUG}" == "SAP/jenkins-library" ]]
     then
-       openssl aes-256-cbc -K $encrypted_12c8071d2874_key -iv $encrypted_12c8071d2874_iv -in cfg/id_rsa.enc -out cfg/id_rsa -d
-      ./gh-pages-deploy.sh
+       echo "Found change on master: Deployment of documentation"
+       PRIVATE_KEY="cfg/id_rsa"
+       openssl aes-256-cbc -K $encrypted_12c8071d2874_key -iv $encrypted_12c8071d2874_iv -in cfg/id_rsa.enc -out "${PRIVATE_KEY}" -d
+       chmod 600 "${PRIVATE_KEY}"
+       eval `ssh-agent -s`
+       ssh-add "${PRIVATE_KEY}"
+       git config user.name "Travis CI Publisher"
+       git remote add docu "git@github.com:$TRAVIS_REPO_SLUG.git";
+       git fetch docu gh-pages:gh-pages
+       echo "Pushing to gh-pages of repository $TRAVIS_REPO_SLUG"
+       cd $TRAVIS_BUILD_DIR/documentation
+       docker run --rm -it -v ~/.ssh:/root/.ssh -v ${PWD}:/docs squidfunk/mkdocs-material gh-deploy -v --clean --remote-name docu
```

Comment:
@radsoulbeard : did you check this? I believe this does not work since we mount a subdirectory of the worktree into the container. This means the git-dir (`.git` at the root level of the project) is not visible inside the docker container. I guess any git call as it is performed when calling mkdocs with `gh-deploy` will not succeed since "HE" is not able to see the git-dir. Convince me from the opposit.

We have to volume-mount the full git repo and we have to excute mkdocs inside a sub folder, there is docker run option `-w` (like `-w /docs/documentation`) for doing it like this.